models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php

```diff
@@ -273,6 +273,24 @@ protected function enrichRelation($object, $params, &$classId, &$relation = [])
      */
     public function save($object, $params = [])
     {
+        $context = $params["context"];
+
+
+        if (!DataObject\AbstractObject::isDirtyDetectionDisabled() && method_exists($object, 'isFieldDirty')) {
```

Comment:
why not introduce new interfaces instead of calling `method_exists` everywhere? would be clearer to understand.